### ISSUE TITLE
Invalid IDE0059 when local.ExtensionMethod used as action

### ISSUE THREAD

**Issue by YairHalberstadt** [dotnet/roslyn#42337]
**Version Used**: master as of the 10th march

**Steps to Reproduce**:

```csharp
using System;
public static class Program {
    public static void Main() {
        var x = 42;
        Func<string> a = x.ToStringExt;
        x = 10;
        Console.WriteLine(a());
    }
    
    public static string ToStringExt(this object x) => x.ToString();
}
```

**Expected Behavior**: No warning.

**Actual Behavior**:
IDE0059 unnecessary assignment of a value to 'x'
This is incorrect because `42` will be printed not `10`, so you cannot change the assignment.
Also the suggested codefix 'remove redundant assignment' doesn't compile:

```csharp
using System;
public static class Program {
    public static void Main() {
        int x;
        Func<string> a = () => x.ToString(); // error CS0165: Use of unassigned local variable 'x'
        x = 41;
        Console.WriteLine(a());
    }
}
```

**Comment by mavasani** [comment]
Also reported at https://developercommunity.visualstudio.com/content/problem/808290/intellisense-does-not-recognize-closured-variables.html
